### ISSUE TITLE
@aws-sdk/client-secrets-manager: 'credentials' does not exist in type 'SecretsManagerClientConfigType'

### ISSUE THREAD

**Issue by biraj-pickcel** [aws/aws-sdk-js-v3#5813]
### Checkboxes for prior research

- [X] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest)
- [X] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js).
- [X] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js-v3/issues) and didn't find any solution.

### Describe the bug

- @aws-sdk/client-secrets-manager v3.515.0
- Node v18.18.1
- TypeScript v 5.3.2

Doing the following give a TypeScript error:
```typescript
const client = new SecretsManagerClient({
    region: process.env.AWS_REGION,
    credentials: {
        accessKeyId: "<my-access-key>",
        secretAccessKey: "<my-secret-access-key>",
    },
});
```

The error:
```
  TS2345: Argument of type '[{ region: string; credentials: { accessKeyId: string; secretAccessKey: string; }; }]' is not assignable to parameter of type '[] | [SecretsManagerClientConfig]'.
  Type '[{ region: string; credentials: { accessKeyId: string; secretAccessKey: string; }; }]' is not assignable to type '[SecretsManagerClientConfig]'.
    Object literal may only specify known properties, and 'credentials' does not exist in type 'SecretsManagerClientConfigType'.
```

But if i ignore the error & run the code, it just works. So the issue is in type declarations.

### SDK version number

@aws-sdk/client-secrets-manager@3.515.0

### Which JavaScript Runtime is this issue in?

Node.js

### Details of the browser/Node.js/ReactNative version

v18.18.1

### Reproduction Steps

Just initialise a SecretsManager client.

```typescript
const client = new SecretsManagerClient({
    region: process.env.AWS_REGION,
    credentials: {
        accessKeyId: "<your-access-key>",
        secretAccessKey: "<your-secret-access-key>",
    },
});
```

### Observed Behavior

```
ERROR in /Users/biraj/Desktop/.../src/app/instances/aws/secret-manager.ts
./src/app/instances/aws/secret-manager.ts 17:4-15
[tsl] ERROR in /Users/biraj/Desktop/.../src/app/instances/aws/secret-manager.ts(17,5)
      TS2345: Argument of type '[{ region: string; credentials: { accessKeyId: string; secretAccessKey: string; }; }]' is not assignable to parameter of type '[] | [SecretsManagerClientConfig]'.
  Type '[{ region: string; credentials: { accessKeyId: string; secretAccessKey: string; }; }]' is not assignable to type '[SecretsManagerClientConfig]'.
    Object literal may only specify known properties, and 'credentials' does not exist in type 'SecretsManagerClientConfigType'.
ts-loader-default_e3b0c44298fc1c14
 @ ./src/app/helpers/aws/secret-manager.ts 2:0-55 5:11-22 9:11-22
 @ ./src/app/controllers/secret.ts 2:0-82 9:22-37
 @ ./src/app/graphql/modules/secret/resolver.ts 1:0-71 7:27-37 14:86-98
 @ ./src/app/graphql/modules.ts 21:0-55 59:4-18
 @ ./src/app/graphql/application.ts 2:0-48 5:4-12 6:4-13
 @ ./src/index.ts

webpack 5.89.0 compiled with 1 error in 5436 ms
```

### Expected Behavior

To not say `'credentials' does not exist in type 'SecretsManagerClientConfigType'` and let me compile my code.

**Comment by kuhe** [comment]
what is your version of `@aws-sdk/core`?

`SecretsManagerClientConfigType`'s intersection contains `HttpAuthSchemeInputConfig` which extends `AwsSdkSigV4AuthInputConfig` which contains 

```ts
credentials?: AwsCredentialIdentity | AwsCredentialIdentityProvider;
```

**Comment by garysassano** [comment]
I think the problem is related to this change that was introduced:

- use bundled dist-cjs build (https://github.com/aws/aws-sdk-js-v3/pull/5687) ([5f79e225](https://github.com/aws/aws-sdk-js-v3/commit/5f79e225e32f0b2ed5b432ae4e4108663eb0acfb))

**Comment by biraj-pickcel** [comment]
i didn't install `@aws-sdk/core` when i opened the issue. but now i explicitly installed it & the error's gone. strangely enough, i uninstalled it & it still works. and the version it shows in package-lock.json `3.513.0`. i suppose i should close this issue now?

**Comment by kuhe** [comment]
Yes, I don't think there was any SDK issue here.

**Comment by biraj-pickcel** [comment]
I removed the `node_modules` folder, reverted my package.json & package-lock.json files, and reinstalled `@aws-sdk/client-secrets-manager`. The issue again popped up. I checked in package-lock.json & `@aws-sdk/core@3.451.0`. Note that I didn't install it explicitly since it's supposed to be installed automatically.

Here's all the aws-sdk-js-v3 packages that we're using
```
    "@aws-sdk/client-appsync": "^3.454.0",
    "@aws-sdk/client-cognito-identity-provider": "^3.454.0",
    "@aws-sdk/client-s3": "^3.465.0",
    "@aws-sdk/client-secrets-manager": "^3.523.0",
 ```

And I got the same error for which I opened this issue. Here's the screenshot:
<img width="1440" alt="Screenshot 2024-02-29 at 11 51 53 AM" src="https://github.com/aws/aws-sdk-js-v3/assets/125873401/01f37659-c24e-43b3-b9ce-acd9d974868f">
